docs/rfcs/003-consolidate-core-packages.md

```diff
@@ -0,0 +1,338 @@
+
+1. Fork and clone `atom/atom` and switch to a release branch if necessary
+1. Build Atom and launch it in dev mode
+1. Make desired changes to the code in `packages/tree-view`
+1. Open a PR on `atom/atom` and wait for CI to pass and a maintainer to review it
+1. Work with maintainers to get the PR approved and merged
+
+At this point, the change is merged into Atom and ready for inclusion in the next release.
+
+### Greatly reduced burden for maintainers
+
+Since packages all have their own repositories, this means that we have to watch 91 different repos for issues and pull requests.  This also means that we have to redirect issues filed on `atom/atom` to the appropriate repository when a user doesn't know where it belongs.  Even more importantly, there's not an easy way to prioritize and track issues across the Atom organization without using GitHub Projects.
+
+Also, as mentioned above, there's the added duty of doing the package "version dance" when we merge any new PRs to a package repository: publish the package update, update `package.json` in Atom.  It's very easy to forget to do this and not have community contributions included in the next Atom release!
+
+The more core packages live in `atom/atom`, the less work Atom maintainers have to do overall.
+
+## Explanation
+
+Many of Atom's core packages now live in the core `atom/atom` repository.  To the Atom user, this change will be imperceptible as these packages still show up in the list of Core Packages in the Settings View.  Users can still disable these packages if they want to replace their behavior with other packages.
+
+For maintainers and contributors, there will be less juggling of repositories and no more publishing of updates to these packages with `apm`:
+
+Contributors now clone and build `atom/atom` to work on improvements to core packages.  They will no longer have to use `apm link` in dev mode to test changes they make to packages in the repo's `packages` folder.  Core packages that aren't consolidated still have folders under `packages` with README.md files that point to the home repository for that package.
+
+When a contributor sends a PR to `atom/atom` that only affects files in a folder under `packages`, only the specs for the relevant package folders will be executed using Atom's CI scripts.  This means that a full Atom build will not be required when no Atom Core code is changed in a PR.  Package specs are also now run against all 3 OSes on Atom `master` and release builds.
+
+Atom maintainers no longer have to publish new versions to consolidated core packages and then edit `package.json` to bump the package version in a particular Atom release branch (Stable, Beta, or `master`).  When a PR against a consolidated core package in `atom/atom` is merged, no version number change is required and the changes will immediately be a part of the next release from that branch.
+
+## Drawbacks
```

Comment:
Following up from the discussion in https://github.com/atom/atom/pull/17674#discussion_r202082961:

> I'll add that to the list of things to verify in a possible prototype.

With the prototype in #17686, were you able to try out the forking/linking workflow discussed in https://github.com/atom/atom/pull/17674#discussion_r202082961?

Reply:
Yep, it works as expected in dev mode if you've used `apm link --dev` for the forked package.  In that PR I'm adding the `atom/packages` path to the list of possible package folders in dev mode *after* the user's `~/.atom/dev/packages` folder, meaning that their dev-linked packages get prioritized.

In the case that they do a plain `apm link` of a forked package and run Atom in normal, non-dev mode, their non-dev linked package will take precedence over the bundled package (just as it always did).